tools/system_libs.py

```diff
@@ -183,13 +184,15 @@ def create_ninja_file(input_files, filename, libname, cflags, asflags=None, cust
   if asflags is None:
     asflags = []
 
+  join_cmd = shlex.join if os.name == 'posix' else subprocess.list2cmdline
```

Comment:
Can we just use `subprocess.list2cmdline` in all cases?

Reply:
Maybe we should re-add `shlex_join` to `utils.py` and use that instead of `shlex.join` everywhere.   Effectivly reverting #24111 but with a cleaner `shlex_join` impl?

Reply:
Maybe, but not sure.

As I said in https://github.com/emscripten-core/emscripten/issues/24393#issuecomment-2899562539, it does look like all other usages are for debug purposes only so syntax doesn't matter as much, and for command invocations we normally use `subprocess` module which does the right thing via native APIs automatically, so Ninja usecase seems somewhat special in that regard.

If nothing else pops up, I wouldn't bother generalising it for now.